client/me/purchases/membership-item/index.tsx

```diff
@@ -89,27 +89,23 @@ const MembershipType = ( { subscription }: { subscription: MembershipSubscriptio
 const Icon = ( { subscription }: { subscription: MembershipSubscription } ) => {
 	const [ hasError, setErrors ] = useState( false );
 	const [ site, setSite ] = useState( null );
-	const [ loadData, setLoadData ] = useState( true );
-
-	async function fetchData() {
-		const data = await fetch(
-			'https://public-api.wordpress.com/rest/v1.1/sites/' + subscription.site_id
-		);
-
-		data
-			.json()
-			.then( ( data ) => {
-				setSite( data );
-				setLoadData( false );
-			} )
-			.catch( ( err ) => setErrors( err ) );
-	}
 
 	useEffect( () => {
-		if ( loadData ) {
-			fetchData();
+		async function fetchData() {
+			const data = await fetch(
+				'https://public-api.wordpress.com/rest/v1.1/sites/' + subscription.site_id
```

Comment:
This is great! The only issue is that if `subscription.site_id` changes, it will not fetch this data again. That may not be a problem, but I think it's worth handling just in case. I'll add three small suggested changes which should handle this.

```suggestion
				'https://public-api.wordpress.com/rest/v1.1/sites/' + siteId
```